addons/pos_self_order/controllers/kiosk_entry.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+import werkzeug
+
+from odoo import http
+from odoo.http import request
+from odoo.addons.pos_self_order.controllers.utils import reduce_privilege
+
+
+class PosSelfKiosk(http.Controller):
+    @http.route(["/kiosk/<config_id>", "/kiosk/<config_id>/<path:subpath>"], auth="public", website=True, sitemap=True)
+    def pos_self_order_kiosk_start(self, config_id=None, access_token=None):
+        lang = request.httprequest.cookies.get('frontend_lang')
+        if lang != None:
+            language = request.env["res.lang"].sudo().search([("iso_code", "=", lang)], limit=1)
+            if language:
+                lang_code = request.env['res.lang']._lang_get_code(language.code)
+                request.lang = language
+                request.update_context(lang=lang_code)
+
+        pos_config_sudo = request.env["pos.config"].sudo().search([
+            ("id", "=", config_id),
+            ('access_token', '=', access_token)], limit=1)
+
+        if not access_token or not config_id or not pos_config_sudo:
```

Comment:
We could do a check before searching the database for the access_token and the config_id and then this check here only for the pos_config_sudo. It would reduce call to the database when we know it wont be fructuous.